meteor/sofie-server/data-access/repositories/test/mongo-part-repository.spec.ts

```diff
@@ -0,0 +1,174 @@
+		it('calls deletion of pieces, matching amount of parts', async () => {
+			const mongoConverter: MongoEntityConverter = mock(MongoEntityConverter)
+			const pieceRepository: PieceRepository = mock<PieceRepository>()
+			const segmentId: string = 'someSegmentId'
+			const parts: Part[] = [createPart({ segmentId: segmentId }), createPart({ segmentId: segmentId })]
+			const pieces: Piece[] = [createPiece({}), createPiece({}), createPiece({})]
+			await testDatabase.populateDatabaseWithParts(parts)
+
+			when(mongoConverter.convertParts(anything())).thenReturn(parts)
+			when(pieceRepository.getPieces(anything())).thenResolve(pieces)
+			const testee: PartRepository = createTestee({
+				mongoConverter: mongoConverter,
+				pieceRepository: pieceRepository,
+			})
+
+			await testee.deletePartsForSegment(segmentId)
+
+			verify(pieceRepository.deletePiecesForPart(anyString())).times(parts.length)
+		})
+
+		it('throws exception, when nonexistent segmentId is given', async () => {
+			const expectedErrorMessageFragment: string = 'Expected to delete one or more parts'
+			const mongoConverter: MongoEntityConverter = mock(MongoEntityConverter)
+			const nonExistingId: string = 'nonExistingId'
+			const part: Part = createPart({})
+			await testDatabase.populateDatabaseWithParts([part])
+
+			when(mongoConverter.convertParts(anything())).thenReturn([])
+			const testee: PartRepository = createTestee({
+				mongoConverter: mongoConverter,
+			})
+
+			const action = async () => testee.deletePartsForSegment(nonExistingId)
+
+			await expect(action).rejects.toThrow(DeletionFailedException)
+			await expect(action).rejects.toThrow(expectedErrorMessageFragment)
+		})
+
+		it('does not deletes any pieces, when nonexistent segmentId is given', async () => {
+			const mongoConverter: MongoEntityConverter = mock(MongoEntityConverter)
+			const nonExistingId: string = 'nonExistingId'
+			const part = createPart({})
+			await testDatabase.populateDatabaseWithParts([part])
+			const db = testDatabase.getDatabase()
+
+			when(mongoConverter.convertParts(anything())).thenReturn([])
+			const testee = createTestee({
+				mongoConverter: mongoConverter,
+			})
+			const action = async () => testee.deletePartsForSegment(nonExistingId)
+
+			await expect(action).rejects.toThrow(DeletionFailedException)
+			await expect(db.collection(COLLECTION_NAME).countDocuments()).resolves.toBe(1)
+		})
+
+		// eslint-disable-next-line jest/expect-expect
+		it('deletes pieces before parts', async () => {
+			const mongoDb: MongoDatabase = mock(MongoDatabase)
+			const pieceRepository: PieceRepository = mock<PieceRepository>()
+			const mongoConverter = mock(MongoEntityConverter)
+			const segmentId: string = 'someSegmentId'
+			const part: Part = createPart({ segmentId: segmentId })
+			await testDatabase.populateDatabaseWithParts([part])
+			const db: Db = testDatabase.getDatabase()
+			const collection = db.collection(COLLECTION_NAME)
+			const spied = spy(collection)
```

Comment:
`spied` does say much. Perhaps `spiedCollection`?

Reply:
Suggestion Applied.

Reply:
Did you push the suggestion? The variable still says `spied`?

Reply:
Missed that i had multiple of the same named constant. All instances (afaik) has now been renamed.